File: app/src/main/java/com/ivanovsky/passnotes/data/entity/UsedFile.java
package com.ivanovsky.passnotes.data.entity;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;
import androidx.room.TypeConverter;

import com.ivanovsky.passnotes.data.repository.file.FSType;

import org.apache.commons.lang3.builder.EqualsBuilder;
import org.apache.commons.lang3.builder.HashCodeBuilder;

@Entity(tableName = "used_file")
public class UsedFile {

	@PrimaryKey(autoGenerate = true)
	@ColumnInfo(name = "id")
	private int id;

	@ColumnInfo(name = "added_time")
	private long addedTime;

	@ColumnInfo(name =  "last_access_time")
	private Long lastAccessTime;

	@ColumnInfo(name = "file_path")
	private String filePath;

	@ColumnInfo(name = "file_uid")
	private String fileUid;

	@ColumnInfo(name = "fs_type")
	private FSType fsType;

	public UsedFile() {
	}

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public long getAddedTime() {
		return addedTime;
	}

	public void setAddedTime(long addedTime) {
		this.addedTime = addedTime;
	}

	public String getFilePath() {
		return filePath;
	}

	public void setFilePath(String filePath) {
		this.filePath = filePath;
	}

	public Long getLastAccessTime() {
		return lastAccessTime;
	}

	public void setLastAccessTime(Long lastAccessTime) {
		this.lastAccessTime = lastAccessTime;
	}

	public String getFileUid() {
		return fileUid;
	}

	public void setFileUid(String fileUid) {
		this.fileUid = fileUid;
	}

	public FSType getFsType() {
		return fsType;
	}

	public void setFsType(FSType fsType) {
		this.fsType = fsType;
	}

	@Override
	public boolean equals(Object o) {
		if (this == o) return true;

		if (o == null || getClass() != o.getClass()) return false;

		UsedFile usedFile = (UsedFile) o;

		return new EqualsBuilder()
				.append(id, usedFile.id)
				.append(addedTime, usedFile.addedTime)
				.append(lastAccessTime, usedFile.lastAccessTime)
				.append(filePath, usedFile.filePath)
				.append(fileUid, usedFile.fileUid)
				.append(fsType, usedFile.fsType)
				.isEquals();
	}

	@Override
	public int hashCode() {
		return new HashCodeBuilder(17, 37)
				.append(id)
				.append(addedTime)
				.append(lastAccessTime)
				.append(filePath)
				.append(fileUid)
				.append(fsType)
				.toHashCode();
	}

}

File: app/src/main/kotlin/com/ivanovsky/passnotes/data/entity/UsedFile.kt
package com.ivanovsky.passnotes.data.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import androidx.room.TypeConverters
import com.ivanovsky.passnotes.data.crypto.entity.BiometricData
import com.ivanovsky.passnotes.data.repository.db.converters.BiometricDataTypeConverter
import com.ivanovsky.passnotes.data.repository.db.converters.KeyTypeConverter

@Entity(tableName = "used_file")
@TypeConverters(KeyTypeConverter::class, BiometricDataTypeConverter::class)
data class UsedFile(
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "id")
    val id: Int? = null,

    @ColumnInfo(name = "fs_authority")
    val fsAuthority: FSAuthority,

    @ColumnInfo(name = "file_path")
    val filePath: String,

    @ColumnInfo(name = "file_uid")
    val fileUid: String,

    @ColumnInfo(name = "file_name")
    val fileName: String,

    @ColumnInfo(name = "added_time")
    val addedTime: Long,

    @ColumnInfo(name = "last_access_time")
    val lastAccessTime: Long? = null,

    @ColumnInfo(name = "key_type")
    val keyType: KeyType,

    @ColumnInfo(name = "key_file_fs_authority")
    val keyFileFsAuthority: FSAuthority? = null,

    @ColumnInfo(name = "key_file_path")
    val keyFilePath: String? = null,

    @ColumnInfo(name = "key_file_uid")
    val keyFileUid: String? = null,

    @ColumnInfo(name = "key_file_name")
    val keyFileName: String? = null,

    @ColumnInfo(name = "biometric_data")
    val biometricData: BiometricData? = null
)